src/eureka/S3_data_reduction/plots_s3.py

```diff
@@ -565,3 +568,307 @@ def median_frame(data, meta):
+    plt.tight_layout()
+    fname = (f'figs{os.sep}fig3503_file{file_number}_int{int_number}_Centroid_Fit'
+             + figure_filetype)
+    plt.savefig(meta.outputdir + fname, dpi=250)
+    if not meta.hide_plots:
+        plt.pause(0.2)
+
+
+def add_colorbar(im, aspect=20, pad_fraction=0.5, **kwargs):
+    """
+    Add a vertical color bar to an image plot.
+    Taken from:
+    https://stackoverflow.com/questions/18195758/set-matplotlib-colorbar-size-to-match-graph
+    """
+    divider = axes_grid1.make_axes_locatable(im.axes)
+    width = axes_grid1.axes_size.AxesY(im.axes, aspect=1./aspect)
+    pad = axes_grid1.axes_size.Fraction(pad_fraction, width)
+    current_ax = plt.gca()
+    cax = divider.append_axes("right", size=width, pad=pad)
+    plt.sca(current_ax)
+    return im.axes.figure.colorbar(im, cax=cax, **kwargs)
+
+
+def phot_2d_frame(meta, m, i, data):
+    """
+    Plots the 2D frame together with the centroid position and apertures.
+    """
+    plt.figure(3306, figsize=(8, 3))
+    plt.clf()
+    plt.suptitle('2D frame with centroid and apertures')
+
+    flux, centroid_x, centroid_y = data.flux[i], data.centroid_x[i], data.centroid_y[i]
+
+    xmin, xmax = data.flux.x.min().values-meta.xwindow[0], data.flux.x.max().values-meta.xwindow[0]
+    ymin, ymax = data.flux.y.min().values-meta.ywindow[0], data.flux.y.max().values-meta.ywindow[0]
+
+    im = plt.imshow(flux, vmin=0, vmax=5e3, origin='lower', aspect='equal',
+                    extent=[xmin, xmax, ymin, ymax])
+    plt.scatter(centroid_x, centroid_y, marker='x', s=25, c='r', label='centroid')
+    plt.title('Full Frame')
+    plt.ylabel('y pixels')
+    plt.xlabel('x pixels')
+
+    circle1 = plt.Circle((centroid_x, centroid_y), meta.photap, color='r',
+                         fill=False, lw=3, alpha=0.7, label='target aperture')
+    circle2 = plt.Circle((centroid_x, centroid_y), meta.skyin, color='w',
+                         fill=False, lw=4, alpha=0.8, label='sky aperture')
+    circle3 = plt.Circle((centroid_x, centroid_y), meta.skyout, color='w',
+                         fill=False, lw=4, alpha=0.8)
+    plt.gca().add_patch(circle1)
+    plt.gca().add_patch(circle2)
+    plt.gca().add_patch(circle3)
+    add_colorbar(im, label='Flux (electrons)')
+    plt.xlim(0, flux.shape[1])
+    plt.ylim(0, flux.shape[0])
+    plt.xlabel('x pixels')
+    plt.ylabel('y pixels')
+
+    plt.legend()
+    plt.tight_layout()
+    file_number = str(m).zfill(int(np.floor(np.log10(meta.num_data_files))+1))
+    int_number = str(i).zfill(int(np.floor(np.log10(meta.n_int))+1))
+
+    fname = (f'figs{os.sep}fig3306_file{file_number}_int{int_number}_2D_Frame'
+             + figure_filetype)
+    plt.savefig(meta.outputdir + fname, dpi=250, tight_layout=True)
+    if not meta.hide_plots:
+        plt.pause(0.2)
+
+
+def phot_2d_frame_zoom(meta, m, i, data):
+    """
+    Plots the 2D frame together with the centroid position and apertures.
```

Comment:
Please explain here how this function differs from the one above

Reply:
Also, I feel like this function could be merged with the one above with a small number of if/else statements